test/uk/gov/hmrc/plasticpackagingtax/returns/forms/RecycledPlasticWeightSpec.scala

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.plasticpackagingtax.returns.forms
+
+import org.scalatest.matchers.must.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+import play.api.data.FormError
+import uk.gov.hmrc.plasticpackagingtax.returns.forms.ManufacturedPlasticWeight.{
+  invalidFormatError,
+  maxTotalKg,
+  totalKg,
+  totalKgBelowThreshold,
+  weightEmptyError
+}
+
+class RecycledPlasticWeightSpec extends AnyWordSpec with Matchers {
+
+  "Recycled plastic weight validation rules" should {
+
+    "return success" when {
+
+      "total is valid" in {
+
+        val input = Map(totalKg -> "100")
+
+        val form = ManufacturedPlasticWeight.form().bind(input)
+        form.errors.size mustBe 0
+      }
+
+      "is 0" in {
+
+        val input = Map(totalKg -> "0")
+
+        val form = ManufacturedPlasticWeight.form().bind(input)
+        form.errors.size mustBe 0
+      }
+
+      "is exactly max allowed amount" in {
+
+        val input =
+          Map(totalKg -> maxTotalKg.toString)
+
+        val form = ManufacturedPlasticWeight.form().bind(input)
+        form.errors.size mustBe 0
+      }
+
+      "contains spaces before and after value" in {
```

Comment:
Out of curiosity do we do this elsewhere? 👀

Reply:
we should have that on all fields, but definitely something to check, I'll create a story to go through all fields and double check the validation is consistent